File: project_euler/p009.py
#	
#	Project Euler 
#	Problem #9
#
#	Dylan Bowers
#
#	https://projecteuler.net/problem=9
#	

def p009():
	for i in range(1, 1001):
		for q in range(1, 1001 - (500 - i)):
			r = 1000 - (i + q)
			if pow(i, 2) + pow(q, 2) == pow(r, 2):
				return i * q * r

if __name__ == '__main__':
	print(p009())

File: project_euler/p009.cpp
/*

	Project Euler 
	Problem #9

	Dylan Bowers

	https://projecteuler.net/problem=9
	
*/

#include <iostream>
#include <stdio.h>
#include <math.h>
using namespace std;
#define LIMIT (1000)

int p009() {
	for(int i = 1; i <= LIMIT; i++) {
		for(int q = 1; q <= LIMIT - (500 - i); q++) {
			int r = 1000 - (i + q);
			if(i * i + q * q == r * r) {
				return i * q * r;
			}
		}
	}
	return 0;
}

int main() {
	printf("%d\n", p009());
}